File: web/src/layout/DocsXml.ts
//
// DocsXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'khrysalis/dist/views/html'
import { customViewSetDelegate } from 'khrysalis/dist/views/CustomView.actual'
import htmlForDefault from './docs.html'
//! Declares com.lightningkite.khrysaliswebsite.layouts.DocsXml
export class DocsXml {
    xmlRoot!: HTMLElement;
    filter!: HTMLInputElement;
    articles!: HTMLDivElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.filter = getViewById<HTMLInputElement>(view, "filter");
        this.articles = getViewById<HTMLDivElement>(view, "articles");
        return view
    }
}

File: android/src/main/java/com/lightningkite/khrysaliswebsite/layouts/DocsXml.kt
//
// DocsXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysaliswebsite.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysaliswebsite.R

class DocsXml {

    lateinit var filter: EditText
    lateinit var articles: com.lightningkite.khrysalis.views.android.VerticalRecyclerView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.docs, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        filter = view.findViewById<EditText>(R.id.filter)
        articles = view.findViewById<com.lightningkite.khrysalis.views.android.VerticalRecyclerView>(R.id.articles)
        
        
        
        return view
    }
}